editor/app/src/main/java/ca/ubc/eml/soiltopargraphy/editor/ui/infopanel/ImagePanelFragment.kt

```diff
@@ -0,0 +1,202 @@
+import android.util.Log
+import java.io.File
+import java.io.FileOutputStream
+import java.io.IOException
+import java.io.OutputStream
+import java.net.URI
+import java.text.SimpleDateFormat
+import java.time.Instant
+import java.time.ZoneId
+import java.time.ZoneOffset
+import java.time.format.DateTimeFormatter
+import java.util.*
+
+
+/**
+ * Fragment where user uploads image from library or take image
+ */
+
+
+class ImagePanelFragment : Fragment() {
+    lateinit var photoPath: String
+    lateinit var photoUri: Uri
+    private var mViewModel: ImagePanelViewModel? = null
+    private lateinit var mImageView: ImageView
+
+    fun onImageAddCameraButtonClick() {
+        //check permission
+        var camerapermission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.CAMERA)
+        var writeExternalPermission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        val requestCode = 0
+        if (camerapermission != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this.requireActivity(),
+                    arrayOf(Manifest.permission.CAMERA),
+                    requestCode)
+        }
+        if (writeExternalPermission != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this.requireActivity(),
+                    arrayOf(Manifest.permission.WRITE_EXTERNAL_STORAGE),
+                    requestCode)
+        }
+        camerapermission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.CAMERA)
+        writeExternalPermission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.WRITE_EXTERNAL_STORAGE)
+        if (camerapermission == PackageManager.PERMISSION_GRANTED && writeExternalPermission == PackageManager.PERMISSION_GRANTED)
+            addImageFromCamera()
+
+    }
+
+    //add image via gallery
+    fun onImageAddGalleryButtonClick() {
+        //check permission
+        var permission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.READ_EXTERNAL_STORAGE)
+        val requestCode = 0
+        if (permission != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(this.requireActivity(),
+                    arrayOf(Manifest.permission.READ_EXTERNAL_STORAGE),
+                    requestCode)
+        }
+        permission = ContextCompat.checkSelfPermission(this.context!!, Manifest.permission.READ_EXTERNAL_STORAGE)
+        if (permission == PackageManager.PERMISSION_GRANTED)
+            addImageFromGallery()
+    }
+
+    fun addImageFromGallery() {
+        val getIntent = Intent(Intent.ACTION_GET_CONTENT)
+        getIntent.setType("image/*")
+        startActivityForResult(getIntent, IMAGE_PICK_CODE)
+    }
+
+    fun createImageFile(): File? {
+        val fileName = "MyPicture"
+        val storageDir = context?.getExternalFilesDir(Environment.DIRECTORY_PICTURES)
```

Comment:
Could we use [internal storage instead of external storage](https://developer.android.com/training/data-storage/files#InternalVsExternalStorage)? That way we won't need the external storage permissions.